client/src/searchPage/filterPanel/FilterPanelController.js

```diff
@@ -2,6 +2,15 @@ import angular from 'angular';
 
 
 export default angular.module('searchPage')
-    .controller('FilterPanelController', ['filterPanelService', function FilterPanelController(filterPanelService) {
-        this.filteredArticles = filterPanelService.getFilteredArticles();
+    .controller('FilterPanelController', ['$state', function FilterPanelController($state) {
+        this.filters = ($state.params.filters && $state.params.filters.split(',').map(Number)) || [];
```

Comment:
This function works with bugs. Please. ensure correct working. Steps to reproduce:
1. go to search page:
2. Click filters in order News, Topics, Journals, Taxes everything looks good:
3. Click on other filter group resets previous filters.

Reply:
Fixed. Good job spotting that one